### PR TITLE
Directory Structure Implementation

### PR DESCRIPTION
**Pull request by JeyJeyGao**
1. support dir package
2. supoort notation-go/config package

Test:
tested with the workflow mentioned in [hello-signing](https://github.com/notaryproject/notation/blob/main/docs/hello-signing.md)
1. `notation cert generate-test --default "wabbit-networks.io"`
2. `notation --plain-http sign $IMAGE`
3. `notation --plain-http list $IMAGE`
4. `notation --plain-http verify $IMAGE`
5. `notation cache list`
6. `notation plugin list`
7. `notation pull $IMAGE`
8. `notation push $IMAGE`
9. `notation key list`
10. `notation cert add --name "wabbit-networks.io" ~/.config/notation/localkeys/wabbit-networks.io.crt`

Signed-off-by: Junjie Gao <43160897+JeyJeyGao@users.noreply.github.com>

### CODE REVIEW

File: cmd/notation/cert_gen.go
Comment (binbin-li) [inline cmd/notation/cert_gen.go:49]: I feel passing in 2 vars: `name` and `extention` might be more clear than constructing a name with extention manually.

Comment (JeyJeyGao) [reply cmd/notation/cert_gen.go]: Done. Thanks!

File: pkg/auth/native_store.go
Comment (patrickzheng200) [inline pkg/auth/native_store.go:33]: Looks like this is a test log? We should remove it.

File: pkg/auth/native_store.go
Comment (patrickzheng200) [inline pkg/auth/native_store.go:54]: Same here?

File: pkg/auth/native_store.go
Comment (JeyJeyGao) [reply pkg/auth/native_store.go]: Deleted.

File: pkg/auth/native_store.go
Comment (JeyJeyGao) [reply pkg/auth/native_store.go]: Deleted.

**Review by binbin-li — APPROVED** [review]
lgtm

**Review by chloeyin — APPROVED** [review]
lgtm

**Review by patrickzheng200 — APPROVED** [review]
lgtm

**Comment by dtzar** [comment]
Per the [PR review criteria](https://github.com/notaryproject/notation/discussions/263) @notaryproject/notation-sub-group-maintainers - this is past a week old and has two approvals from official code-reviewers, so it is eligible for merge now.

File: pkg/config/once.go
Comment (shizhMSFT) [inline pkg/config/once.go:35]: Curious why we don't call `github.com/notaryproject/notation-go/config` directly and rename this package to something like `configutil`?

Comment (JeyJeyGao) [reply pkg/config/once.go]: Thanks for your suggestion.

File: cmd/notation/cert.go
Comment (shizhMSFT) [inline cmd/notation/cert.go:11]: Let's keep `github.com/notaryproject/notation-go/config` as `config`, and rename `github.com/notaryproject/notation/pkg/config` as `github.com/notaryproject/notation/pkg/configutil`.

**Review by shizhMSFT — CHANGES_REQUESTED** [review]

File: cmd/notation/cert.go
Comment (JeyJeyGao) [reply cmd/notation/cert.go]: Done and tested again.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**
